# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/fake/flite/const-assignment.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./tools/find_sts_main.c.orig	2002-12-26 18:18:30.000000000 +0100
+++ ./tools/find_sts_main.c	2004-08-03 17:41:07.686245736 +0200
@@ -93,14 +93,14 @@
 			lpc->frames[i],lpc->num_channels,
 			resd,
 			size);
-	sts[i].size = size;
+	*(int*)(&sts[i].size) = size;
 	sts[i].frame = cst_alloc(unsigned short,lpc->num_channels-1);
 	for (j=1; j < lpc->num_channels; j++)
-	    sts[i].frame[j-1] = (unsigned short)
+	    *(unsigned short*)(&sts[i].frame[j-1]) = (unsigned short)
 		(((lpc->frames[i][j]-lpc_min)/lpc_range)*65535);
 	sts[i].residual = cst_alloc(unsigned char,size);
 	for (j=0; j < size; j++)
-	    sts[i].residual[j] = cst_short_to_ulaw((short)resd[j]);
+	    *(int*)(&sts[i].residual[j]) = cst_short_to_ulaw((short)resd[j]);
 	start = end;
     }
 
